sw/host/rom_ext_image_tools/signer/image/src/manifest.rs

```diff
@@ -1,111 +1,88 @@
 // Copyright lowRISC contributors.
 // Licensed under the Apache License, Version 2.0, see LICENSE for details.
 // SPDX-License-Identifier: Apache-2.0
-//
-// ---------- W A R N I N G: A U T O - G E N E R A T E D   C O D E !! ----------
-// PLEASE DO NOT HAND-EDIT THIS FILE. IT HAS BEEN AUTO-GENERATED WITH THE
-// FOLLOWING COMMAND:
-// util/rom-ext-manifest-generator.py
-//     --input-dir=sw/device/silicon_creator/rom_exts
-//     --output-dir=<destination dir>
-//     --output-files=rust
 
-pub struct ManifestField {
-    pub offset: usize,
-    pub size_bytes: usize,
+//! Structs for reading and writing manifests of flash boot stage images.
+//!
+//! Note: The structs below must match the definitions in
+//! sw/device/silicon_creator/lib/manifest.h.
+
+#![deny(warnings)]
+#![deny(unused)]
+#![deny(unsafe_code)]
+
+use std::mem::size_of;
+
+use memoffset::offset_of;
+use zerocopy::AsBytes;
+use zerocopy::FromBytes;
+
+// Currently, these definitions must be updated manually but they can be
+// generated using the following commands (requires bindgen):
+//   cargo install bindgen
+//   cd "${REPO_TOP}"
+//   bindgen --allowlist-type manifest_t --allowlist-var "MANIFEST_.*" \
+//      --no-doc-comments --no-layout-tests \
+//      sw/device/silicon_creator/lib/manifest.h \
+//      -- -I./ -Isw/device/lib/base/freestanding
```

Comment:
I think we need to have some documentation written about how and when we use bindgen.  Generally, I object to "run an external tool and paste into code".  I think this is fine for now since the Manifest struct is supposed to be stable and we wouldn't expect this to change much.

I'm not suggesting you change anything here - just that we're going to need to have some written guidance for bindgen soon.

Reply:
Personally I'm ok with checking in generated code iff CI can regenerate it and check it's correct byte for byte